packages/lib/src/components/ClickToPay/ClickToPay.tsx

```diff
@@ -0,0 +1,123 @@
+        this.clickToPayService?.initialize().catch(error => {
+            this.handleError(new AdyenCheckoutError('ERROR', error.toString(), { cause: error }));
+        });
+
+        if (!this.clickToPayService) {
+            console.warn('ClickToPay not initialized - Likely the payment method is not configured or its configuration is missing');
+        }
+    }
+
+    get isValid() {
+        return true;
+    }
+
+    get browserInfo() {
+        return collectBrowserInfo();
+    }
+
+    public formatData(): ClickToPayPaymentData {
+        const { srcScheme, srcCorrelationId, srcTokenReference, srcCheckoutPayload, srcDigitalCardId } = this.state.data;
+        return {
+            paymentMethod: {
+                type: ClickToPayElement.type,
+                ...(srcScheme && { srcScheme }),
+                ...(srcCorrelationId && { srcCorrelationId }),
+                ...(srcTokenReference && { srcTokenReference }),
+                ...(srcCheckoutPayload && { srcCheckoutPayload }),
+                ...(srcDigitalCardId && { srcDigitalCardId })
+            },
+            browserInfo: this.browserInfo,
+            origin: !!window && window.location.origin
+        };
+    }
+
+    protected formatProps(props: ClickToPayElementProps) {
+        return {
+            ...props,
+            disableOtpAutoFocus: props.disableOtpAutoFocus || false,
+            shopperEmail: props.shopperEmail || props?._parentInstance?.options?.session?.shopperEmail,
+            telephoneNumber: props.telephoneNumber || props?._parentInstance?.options?.session?.telephoneNumber,
+            locale: props.locale || props.i18n?.locale?.replace('-', '_')
+        };
+    }
+
+    /**
+     * Method used to let the merchant know if the shopper have a valid CtP accoubt
+     *
+     * Resolves Promise if the Shopper has cookies OR has valid CtP account
+     * Rejects Promise if account isn't found or if Login screen is triggered
+     */
+    public async isAvailable(): Promise<void> {
+        if (!this.clickToPayService) {
+            return Promise.reject();
+        }
+
+        if (this.clickToPayService.shopperAccountFound) {
+            return Promise.resolve();
+        }
+
+        return new Promise((resolve, reject) => {
+            this.clickToPayService.subscribeOnStateChange((state: CtpState) => {
+                if (this.clickToPayService.shopperAccountFound) resolve();
+                if (state === CtpState.NotAvailable || state === CtpState.Login || state === CtpState.Idle) reject();
+            });
+        });
+    }
+
+    private handleClickToPaySubmit = (payload: ClickToPayCheckoutPayload) => {
+        this.setState({ data: { ...payload }, valid: {}, errors: {}, isValid: true });
+        this.submit();
+    };
+
+    render() {
+        return (
+            <CoreProvider i18n={this.props.i18n} loadingContext={this.props.loadingContext} resources={this.resources}>
+                <ClickToPayProvider
```

Comment:
Is there any value in using the `ClickToPayWrapper` here, like we do in the `Card` component? 
Like, consistency of implementation?

Reply:
ok.